frontend/src/pages/ExecutionList.test.tsx

```diff
@@ -0,0 +1,194 @@
+  listOperationOpts.setMaxResultSize(10);
+  const getArtifactsRequest = new GetArtifactsRequest();
+  getArtifactsRequest.setOptions(listOperationOpts),
+    beforeEach(() => {
+      getExecutionTypesSpy.mockImplementation(() => {
+        const executionType = new ExecutionType();
+        executionType.setId(6);
+        executionType.setName('String');
+        const response = new GetExecutionTypesResponse();
+        response.setExecutionTypesList([executionType]);
+        return Promise.resolve(response);
+      });
+
+      getExecutionsSpy.mockImplementation(() => {
+        const executions = mockNExecutions(5);
+        const response = new GetExecutionsResponse();
+        response.setExecutionsList(executions);
+        return Promise.resolve(response);
+      });
+    });
+
+  function generateProps(): PageProps {
+    return TestUtils.generatePageProps(
+      ExecutionList,
+      { pathname: RoutePage.EXECUTIONS } as any,
+      '' as any,
+      historyPushSpy,
+      updateBannerSpy,
+      updateDialogSpy,
+      updateToolbarSpy,
+      updateSnackbarSpy,
+    );
+  }
+
+  function mockNExecutions(n: number) {
+    let executions: Execution[] = [];
+    for (let i = 1; i <= n; i++) {
+      const execution = new Execution();
+      const pipelineValue = new Value();
+      const pipelineName = `pipeline ${i}`;
+      pipelineValue.setStringValue(pipelineName);
+      execution.getCustomPropertiesMap().set('pipeline_name', pipelineValue);
+      const executionValue = new Value();
+      const executionName = `test execution ${i}`;
+      executionValue.setStringValue(executionName);
+      execution.getPropertiesMap().set('name', executionValue);
+      execution.setName(executionName);
+      execution.setId(i);
+      executions.push(execution);
+    }
+    return executions;
+  }
+
+  it('renders one execution', async () => {
+    getExecutionsSpy.mockImplementation(() => {
+      const executions = mockNExecutions(1);
+      const response = new GetExecutionsResponse();
+      response.setExecutionsList(executions);
+      return Promise.resolve(response);
+    });
+    render(
+      <CommonTestWrapper>
+        <ExecutionList {...generateProps()} />
+      </CommonTestWrapper>,
+    );
+
+    await waitFor(() => {
+      expect(getExecutionTypesSpy).toHaveBeenCalledTimes(1);
+      expect(getExecutionsSpy).toHaveBeenCalledTimes(1);
+    });
+
+    screen.getByText('test execution 1');
+  });
+
+  it('displays footer with "10" as default value', async () => {
+    render(
+      <CommonTestWrapper>
+        <ExecutionList {...generateProps()} />
+      </CommonTestWrapper>,
+    );
+
+    await waitFor(() => {
+      expect(getExecutionTypesSpy).toHaveBeenCalledTimes(1);
+      expect(getExecutionsSpy).toHaveBeenCalledTimes(1);
+    });
+
+    screen.getByText('Rows per page:');
+    screen.getByText('10');
+  });
+
+  it('it able to see the 20th artifact if "Rows per page" is chanegd to 20', async () => {
```

Comment:
NIT: Replace `'it able to see the 20th artifact if "Rows per page" is chanegd to 20'` with `shows 20th artifact if page size is 20`.